modules/margin/src/main/java/com/opengamma/sdk/margin/v3/PortfolioDataFile.java

```diff
@@ -95,6 +98,10 @@ public static PortfolioDataFile of(Path path) {
    * @throws UncheckedIOException if an IO error occurs
    */
   public static PortfolioDataFile ofCombined(List<Path> paths) {
+    if (paths.stream().anyMatch(Files::notExists)) {
```

Comment:
You could build a nicer error message if you filtered on `Files::notExists` to get a collection of all non-existent files.

    List<Path> missingFiles = paths.stream().filter(Files::notExists).collect(toList());
    if (!missingFiles.isEmpty()) {
      String missingFileNames = missingFiles.stream().map(Object::toString).collect(joining(", "))
      throw new...